pkg/apiserver/authenticator/authn.go

```diff
@@ -58,30 +59,33 @@ type AuthenticatorConfig struct {
 
 // New returns an authenticator.Request or an error that supports the standard
 // Kubernetes authentication mechanisms.
-func New(config AuthenticatorConfig) (authenticator.Request, error) {
+func New(config AuthenticatorConfig) (authenticator.Request, *common.SecurityDefinitions, error) {
 	var authenticators []authenticator.Request
+	securityDefinitions := common.NewSecurityDefinition()
 
 	// BasicAuth methods, local first, then remote
 	if len(config.BasicAuthFile) > 0 {
 		basicAuth, err := newAuthenticatorFromBasicAuthFile(config.BasicAuthFile)
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 		authenticators = append(authenticators, basicAuth)
+		securityDefinitions.AddBasicDefinition()
```

Comment:
a custom object with add methods we call repeatedly on the same auth methods seems overly complicated, when all we really want is to know if we should add a basic security definition and bearer security definition